packages/ui-components/src/components/NativeSelect/NativeSelect.component.tsx

```diff
@@ -0,0 +1,254 @@
+/*
+ * SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Juno contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React, { useState, useEffect, ReactNode, ChangeEvent, MouseEvent, FC } from "react"
+
+import { Icon } from "../Icon/Icon.component"
+import { Spinner } from "../Spinner/Spinner.component"
+
+const selectBaseClasses = `
+  jn-w-full
+  jn-bg-theme-select
+  jn-text-theme-high
+  jn-appearance-none
+  jn-text-base
+  jn-pl-4
+  jn-h-[2.375rem]
+  jn-rounded-3px
+  jn-bg-icon-arrow-down
+  jn-bg-right
+  jn-bg-no-repeat
+  focus:jn-outline-none
+  focus:jn-ring-2
+  focus:jn-ring-theme-focus
+  disabled:jn-opacity-50
+`
+
+const wrapperBaseClasses = `jn-relative`
+
+const iconBaseClasses = `
+  jn-absolute
+  jn-flex
+  jn-right-2
+  jn-top-1.5
+  jn-pointer-events-none
+`
+
+const iconDisabledClasses = `jn-opacity-50`
+
+const borderErrorClasses = `jn-border jn-border-theme-error`
+const borderSuccessClasses = `jn-border jn-border-theme-success`
+
+const stateClasses = {
+  loading: `
+    jn-absolute
+    jn-top-0
+    jn-right-0
+    jn-bottom-0
+    jn-left-0
+    jn-text-center
+    jn-bg-theme-select
+    jn-text-theme-high
+    jn-text-base
+    jn-rounded-3px
+    jn-flex
+    jn-flex-col
+    jn-justify-center
+    jn-select-none
+    jn-cursor-not-allowed
+  `,
+  error: `
+    jn-absolute
+    jn-top-0
+    jn-right-0
+    jn-bottom-0
+    jn-left-0
+    jn-text-center
+    jn-bg-theme-select
+    jn-text-theme-high
+    jn-text-base
+    jn-rounded-3px
+    jn-flex
+    jn-flex-col
+    jn-justify-center
+    jn-select-none
+    jn-cursor-not-allowed
+  `,
+  loadingSpinner: `jn-ml-auto jn-mr-auto`,
+  errorIcon: `jn-ml-auto jn-mr-auto`,
+  paddingIconRight: `jn-pr-[3.75rem]`,
+  paddingDefaultRight: `jn-pr-9`,
+}
+
+export interface NativeSelectProps {
```

Comment:
should not be extended here from something like `React.HTMLAttributes<HTMLDivElement>`?

Reply:
Already committed, thanks for the catch. 🚀